zetasql/reference_impl/algebrizer.cc

```diff
@@ -2296,6 +2296,29 @@ Algebrizer::AlgebrizeLimitOffsetScan(const ResolvedLimitOffsetScan* scan) {
   }
 }
 
+absl::StatusOr<std::unique_ptr<RelationalOp>>
+Algebrizer::AlgebrizeTopScan(const ResolvedTopScan* scan) {
+  ZETASQL_RET_CHECK(scan->top() != nullptr);
+
+  ZETASQL_ASSIGN_OR_RETURN(std::unique_ptr<ValueExpr> limit,
+                   AlgebrizeExpression(scan->top()));
+  ZETASQL_ASSIGN_OR_RETURN(std::unique_ptr<ValueExpr> offset,
```

Comment:
Don't we do extra work here? I see a lot of limit and offset words in this function 😄